crates/wit-component/src/encoding.rs

```diff
@@ -886,6 +1039,43 @@ impl<'a> EncodingState<'a> {
                 Some(self.interface)
             }
         }
+
+        impl NestedComponentTypeEncoder<'_, '_> {
+            fn unique_import_name(&mut self, name: &str) -> String {
+                let base = format!("import-type-{name}");
+                let mut name = base.clone();
+                let mut n = 0;
+                while self.imports.contains_key(&name) {
+                    name = format!("{name}{n}");
+                    n += 1;
+                }
+                name
+            }
+        }
+
+        fn import_func_name(f: &Function) -> String {
+            match f.kind {
+                FunctionKind::Freestanding => {
+                    format!("import-func-{}", f.name)
+                }
+
+                // transform `[method]foo.bar` into `import-method-foo-bar` to
+                // have it be a valid kebab-name which can't conflict with
+                // anything else.
```

Comment:
I'm probably forgetting something here, but why can't `import-method-foo-bar` conflict with user identifiers?

Reply:
The imports into this sub-component are all synthesized by wit-component, and freestanding functions are "mangled" with `import-func-` as a prefix where methods are "mangled" with `import-method-` as a prefix, which is intended to remove the possibility of conflict.